### ISSUE TITLE
Variable "num_annotations" in utils.eval.py yields wrong results? 

### ISSUE THREAD

**Issue by nikostsagk** [fizyr/keras-retinanet#1076]
As far as I understand, the `CSVGenerator` creates a generator with the `inputs` and the `targets` which then are further processed by the `Generator` class. In the `Generator` class, function `filter_annotations` discards defective annotations according to [here](https://github.com/fizyr/keras-retinanet/blob/6a246a13407213f6ef273f3447e754aa13035192/keras_retinanet/preprocessing/generator.py#L155).

However, when I run `keras_retinanet/bin/evaluate.py` to evaluate my dataset, `    all_annotations    = _get_annotations(generator)` in [L169](https://github.com/fizyr/keras-retinanet/blob/master/keras_retinanet/utils/eval.py#L169) (and `num_annotations` as an extension) seem to yield all instances (without the empty images), while I was expecting to see way less instances. 

If there is indeed a problem and I haven't overlooked something, this has a direct effect on the mAP calculation.

**Comment by hgaiser** [comment]
> seem to yield all instances (without the empty images), while I was expecting to see way less instances.

I'm not sure what you were expecting to see?

That function (`_get_annotations`) uses the generator to load annotations. It doesn't filter away annotations like the training process does. I'm not sure if we should change this or not, but I don't think it harms mAP computations. Do you suggest it does?

**Comment by nikostsagk** [comment]
You are right, I later found out that I couldn't actually delete an issue. The reason I started digging the code was that results started to seem strange to me.
E.g: for 538 instances, 516 were detected as TP and 1380 as FP. 
Precision and recall is 0.27 and 0.95 respectively, while mAP according to `_compute_ap`: 0.84.
Intuitively, doesn't seem right to me such high mAP with so many FP but I understand that this is not necessarily a code issue.

>  It doesn't filter away annotations like the training process does.

Upon that, annotations that are defective, for example out of the image boundaries, are discarded or fixed to be exactly on the boundary? 
Validation set probably is treated the same. I thought that they are discarded, but apparently when i use the `evaluation.py`, printed images still have the defective boundaries.
